libtelemetry/src/full/java/com/mapbox/android/telemetry/crash/CrashReporterClient.java

```diff
@@ -0,0 +1,165 @@
+package com.mapbox.android.telemetry.crash;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+import com.mapbox.android.core.FileUtils;
+import com.mapbox.android.telemetry.MapboxTelemetry;
+import com.mapbox.android.telemetry.TelemetryListener;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.mapbox.android.core.crashreporter.MapboxUncaughtExceptionHanlder.MAPBOX_CRASH_REPORTER_PREFERENCES;
+import static com.mapbox.android.core.crashreporter.MapboxUncaughtExceptionHanlder.MAPBOX_LAST_CRASH_REPORTER;
+import static com.mapbox.android.core.crashreporter.MapboxUncaughtExceptionHanlder.MAPBOX_PREF_ENABLE_CRASH_REPORTER;
+
+final class CrashReporterClient {
+  private static final String CRASH_REPORTER_CLIENT_USER_AGENT = "mapbox-android-crash";
+  private final Context applicationContext;
+  private final SharedPreferences sharedPreferences;
+  private final MapboxTelemetry telemetry;
+  private final HashSet<String> crashHashSet = new HashSet<>();
+  private File[] crashReports;
+  private int fileCursor;
+
+  @VisibleForTesting
+  CrashReporterClient(@NonNull Context context,
+                      @NonNull SharedPreferences sharedPreferences,
+                      @NonNull MapboxTelemetry telemetry,
+                      File[] crashReports) {
+    this.applicationContext = context;
+    this.sharedPreferences = sharedPreferences;
+    this.telemetry = telemetry;
+    this.crashReports = crashReports;
+    this.fileCursor = 0;
+  }
+
+  static CrashReporterClient create(@NonNull Context context) {
+    SharedPreferences sharedPreferences =
+      context.getSharedPreferences(MAPBOX_CRASH_REPORTER_PREFERENCES, Context.MODE_PRIVATE);
+    return new CrashReporterClient(context, sharedPreferences,
+      new MapboxTelemetry(context, "", CRASH_REPORTER_CLIENT_USER_AGENT), new File[0]);
+  }
+
+  CrashReporterClient load(@NonNull String rootPath) {
+    crashReports = FileUtils.listAllFiles(applicationContext, rootPath);
```

Comment:
Should `fileCursor` be reset to `0` at this point?

Reply:
yeah fair point, i do initialize it to zero and expectation is that `load` is called once but if we'd reuse it to load other batch at the same time then it has to be reset to zero. did not have chance to write tests for it yet but thanks for the note!

Reply:
I'd encapsulate it into a setter for the property